### ISSUE TITLE
angular2 final release ng-bootstrap component not found 404

### ISSUE THREAD

**Issue by faisalloe** [ng-bootstrap/ng-bootstrap#778]
angular2 final release. ng-bootstrap alpha v.5

bootstrap components works on html, but when i tried to import them as following

```
    import {ViewChild} from "@angular/core/src/metadata/di";
    import {NgbDropdown} from "@ng-bootstrap/ng-bootstrap/dropdown/dropdown”;

```
i got following error.

```
    zone.js:136drainMicroTaskQueue @ zone.js:368ZoneTask.invoke @ zone.js:308
    zone.js:1274 GET http://localhost:8080/abc/node_modules/@ng-bootstrap/ng-
bootstrap/bundles/ng-bootstrap.js/dropdown/dropdown.js 404 (Not Found)
```
certainly looks like it is trying to load dropdown from bundles which is not there.. bundle just contains ng-bootstrap.js 'dropdown' is different component outside the bundle. I think it has to some thing with systemjs.config.js can some one help ?

Following is my systemjs.config.js file
```

    (function (global) {
    
      System.config({
        paths: {
          // paths serve as alias
          'npm:': 'node_modules/'
        },
        // map tells the System loader where to look for things
        map: {
          // our app is within the app folder
          app: 'app',
          // angular bundles
          '@angular/core': 'npm:@angular/core/bundles/core.umd.js',
          '@angular/common': 'npm:@angular/common/bundles/common.umd.js',
          '@angular/compiler': 'npm:@angular/compiler/bundles/compiler.umd.js',
          '@angular/platform-browser': 'npm:@angular/platform-browser/bundles/platform-browser.umd.js',
          '@angular/platform-browser-dynamic': 'npm:@angular/platform-browser-dynamic/bundles/platform-browser-dynamic.umd.js',
          '@angular/http': 'npm:@angular/http/bundles/http.umd.js',
          '@angular/router': 'npm:@angular/router/bundles/router.umd.js',
          '@angular/forms': 'npm:@angular/forms/bundles/forms.umd.js',
    
          'rxjs':                       'npm:rxjs',
          'angular2-in-memory-web-api': 'npm:angular2-in-memory-web-api',
          '@ng-bootstrap/ng-bootstrap':'npm:@ng-bootstrap/ng-bootstrap/bundles/ng-bootstrap.js',
        },
        // packages tells the System loader how to load when no filename and/or no extension
        packages: {
          app: {
            main: './main.js',
            defaultExtension: 'js'
          },
          rxjs: {
            defaultExtension: 'js'
          },
          'angular2-in-memory-web-api': {
            main: './index.js',
            defaultExtension: 'js'
          },
          '@ng-bootstrap':{
            defaultExtension: 'js'
          },
          '@ng-bootstrap/ng-bootstrap':{
            defaultExtension: 'js'
          }
        }
      });
    })(this);

```

**Comment by pkozlowski-opensource** [comment]
tl; dr; Don't import from anything other than `@ng-bootstrap/ng-bootstrap` (same as with Angular - you don't want to import from anything below say `@angular/core`).

Long explanation: currently we re-distribute a bundle with all components where the _only_ entry point is  `@ng-bootstrap/ng-bootstrap`. We don't offer imports for individual components with bundles.

You could certainly import individual files directives but then the setup of tooling becomes far more complex (at least with anything other than Webpack 2). The reason for this is that (as Angular) individual files in the npm package are transpiled to ES5 with ES2015 module format. The only tool that understand those natively atm is WebPack 2.

In short: if you don't want to go into complex bundler setup import _only_ from `@ng-bootstrap/ng-bootstrap`. We might have individual npm packages for each and every component but we are not there.

**Comment by faisalloe** [comment]
i tried as following and got error that @ng-bootstrap/ng-bootstrap/index as no exported member 'NgbDropdown'

`import {NgbDropdown} from "@ng-bootstrap/ng-bootstrap”;`

any help ?

**Comment by pkozlowski-opensource** [comment]
In `1.0.0-alpha.5` we don't export individual modules, only `NgbModule`. This is already changed in master via d1492b728b0cf46c90f2bcf8d02d7c80b53d8f4c, will be released as part of `1.0.0-alpha.6` by the end of the week.

For now do import `{NgbModule} from "@ng-bootstrap/ng-bootstrap”;`

**Comment by faisalloe** [comment]
i totally get that, really appreciate your help. but how can i create a variable of type NgbDropDown ? 

// import ngbModule
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';

now i want to do as following, how can i get reference of NgbDropDown

private dropdown: NgbDropDown

**Comment by pkozlowski-opensource** [comment]
> i totally get that, really appreciate your help. but how can i create a variable of type NgbDropDown ? 

Oh, we don't export it from anywhere (I mean - not from any public entry points) currently :-/ 
I guess you are trying to query for a directive instance, right?

I'm going to create a separate issue for this.

**Comment by pkozlowski-opensource** [comment]
Created #779

**Comment by faisalloe** [comment]
yes pls. 

i want to do some thing as mentioned in the answer here

http://stackoverflow.com/questions/39603711/ng-bootstrap-example-calling-method-from-component

**Comment by MaddyDahiya** [comment]
While try to install ng-bootstrao for Angular5 Project using Angular-cli version 6, got an error below - 
D:\Angular>npm install --save @ng-bootstrap/ng-bootstrap
npm ERR! code E418
npm ERR! 418 I'm a teapot: @ng-bootstrap/ng-bootstrap@latest

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ABC\AppData\Roaming\npm-cache\_logs\2018-05-29T03
_18_26_712Z-debug.log
